lib/pdf_attachment_reporter.rb

```diff
@@ -0,0 +1,179 @@
+require 'csv'
+
+class PDFAttachmentReporter
+  POLICY_GROUPS = 'Policy Groups'.freeze
+
+  def initialize(opts = {})
+    @data_path = opts.fetch(:data_path, ENV['HOME'])
+    @first_period_start_date = opts.fetch(:first_period_start_date, Date.parse('2016-01-01'))
+    @last_time_period_days = opts.fetch(:last_time_period_days, 30)
+  end
+
+  #   Notes/limitations of this report
+  #
+  #   - We do not take into account changes in ownership of a document with
+  #   a PDF attachment over time. Instead we take the most recently-published
+  #   version of the document as the correct source of ownership.
+  #
+  #   - A document that has had a PDF published in the past under one name that
+  #   is subsequently replaced by a PDF with another name is counted in the
+  #   report twice. Even though the site shows only one PDF.
+  #
+  #   - All policy group attachments are lumped into one category
+  #
+  #   - Finding the organisation which owns a PDF can be complex. In
+  #   cases where there are multiple owners, we first try to find a parent
+  #   organisation, if we know that all associated organisations are related.
+  #   Otherwise we assume that the first organisation in the list is a best
+  #   guess.
+
```

Comment:
Moved in https://github.com/alphagov/whitehall/pull/3036/commits/dadc7f30af1e68a422e23ec046b6c8f5170b3226#diff-12bf1caacce1f6bb371a9e4909fa749dR1